### PR TITLE
feat: add basecamp integration to cal.com

### LINKED ISSUE #8641: [CAL-1708] Basecamp integration
### Is your proposal related to a problem?
**It is not.  I'd just like to have Cal.com show up in the integrations with Basecamp besides pasting an external link.  Just is better image wise and promotes Cal.com.  The only one there is Calendly.**
<!--
  Provide a clear and concise description of what the problem is.
  **I'd like for Cal.com to show up in the integrations for Basecamp.  So it opens up directly in basecamp instead of an external link right now.  You can see Calendly is in basecamp.**

<sub>[CAL-1708](https://linear.app/calcom/issue/CAL-1708/basecamp-integration)</sub>

### PR DESCRIPTION
**Pull request by Jonathan1599**
## What does this PR do?

Adds basecamp integration to cal.com. Currently works using the basecamp3 API. This PR is WIP as of now. Thinking of adding Google Calendar as a dependency.

Fixes #8641 

**Environment**: Staging(main branch) / Production

Demo: https://drive.google.com/file/d/1OOYtRcw_b0fiNj0vXBLRDJu2IbN6eszj/view?usp=sharing

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] Follow the instructions [here](https://github.com/basecamp/bc3-api#authentication) to get your client_id and client_secret.
- [x] Add the client_id and client_secret to .env.appstore.
- [x]  Install the app and add it to an event.
- [x]  Link your project to the event type. 
- [x]  New bookings should show up in your basecamp schedule.
- [x]  Contact basecamp regarding using press images and logos as mentioned [here](https://github.com/basecamp/api/blob/master/sections/brand_guidelines.md)
## Checklist

/claim #8641

### CODE REVIEW

File: packages/app-store/basecamp3/api/projects.ts
Comment (Jonathan1599) [inline packages/app-store/basecamp3/api/projects.ts:23]: Provides the API url with the user ID in it

File: packages/app-store/basecamp3/lib/constants.ts
Comment (Jonathan1599) [inline packages/app-store/basecamp3/lib/constants.ts:1]: Need this in every request to comply with their API policies [here](https://github.com/basecamp/bc3-api/tree/master#identifying-your-application)

File: .env.appStore.example
Comment (leog) [inline .env.appStore.example:23]: Could you please add a title and description for these newly introduced environment variables? Also add the title to the index at the top. Furthermore, it will also need to point to a Cal.com README section on how to get the values. You can see the other environment values for other apps as examples.

File: packages/app-store/basecamp3/api/add.ts
Comment (leog) [inline packages/app-store/basecamp3/api/add.ts:27]: Is this suppose to be hardcoded? Can it change for some reason? Like for managing different environments for the API. Just double checking.

File: packages/app-store/basecamp3/api/callback.ts
Comment (leog) [inline packages/app-store/basecamp3/api/callback.ts:18]: Same comment about the hardcoded URL, just to be sure we are covered. Also, NIT but would be great if you could use `URLSearchParams` to generate que params from a JSON object, much cleaner.

File: packages/app-store/basecamp3/api/projects.ts
Comment (leog) [inline packages/app-store/basecamp3/api/projects.ts:51]: Why is this URL so different from the ones used to authenticate? is 37signals the company behind basecamp and they differentiate authenticating using their top company auth system and then use specific APIs from the product? Looks weird, that's all.

File: packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx
Comment (leog) [inline packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:17]: Having my doubts about using `fetch` here for an internal API, not sure if it is expected to have apps to do this, and if so, maybe there is an standard way to do it. Summoning @hariombalhara JIC

File: packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx
Comment (leog) [inline packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:59]: Same here cc: @hariombalhara

File: packages/app-store/basecamp3/config.json
Comment (leog) [inline packages/app-store/basecamp3/config.json:5]: NIT: Is there a way you can find a logo that could look OK in both light and dark modes?

File: packages/app-store/basecamp3/lib/constants.ts
Comment (leog) [reply packages/app-store/basecamp3/lib/constants.ts:1]: This could present an issue, as this will not be applicable for self-hosters. I think this requirement should be clearly pointed out in the section of the README where explains how to get environment variable values and shape it as a new environment variable to be very explicit of its importance and to get the correct value for self-hosters.

**Review by leog — CHANGES_REQUESTED** [review]
Great job @Jonathan1599. I left some comments to tackle before it can be merged. Thanks for your contribution.

File: packages/app-store/basecamp3/api/projects.ts
Comment (Jonathan1599) [reply packages/app-store/basecamp3/api/projects.ts:51]: Yep, 37signals for some reason uses launchpad.37signals.com to authorise users for all their applications it seems.

File: README.md
Comment (leog) [inline README.md:473]: Sorry @Jonathan1599, I meant to create a new environment variable along with `BASECAMP3_CLIENT_ID` and `BASECAMP3_CLIENT_SECRET` to have this setup directly, like `BASECAMP3_USER_AGENT`. Don't forget to add it to the `zod.ts` file as well so it can be changed from the apps section in admin settings. Thanks @joeauyeung for the pointer.

**Review by leog — CHANGES_REQUESTED** [review]
Almost there! Thanks for tackling the feedback, a couple more required changes and good to go.

File: README.md
Comment (Jonathan1599) [reply README.md:473]: Yeah, this is a better solutions, thanks! Will implement

File: packages/app-store/basecamp3/config.json
Comment (Jonathan1599) [reply packages/app-store/basecamp3/config.json:5]: Seems like we're only allowed to use logos and screenshots from [here](https://basecamp.com/press). That being said, I think it depends on the theme as it looks fine in light mode.
image
Any idea on how to solve this? Should I change the background colour on the SVG?

File: README.md
Comment (leog) [inline README.md:473]: An example would be nice, like "Cal.com (support@cal.com)" 👍

File: packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx
Comment (hariombalhara) [reply packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:17]: This is a first and unique case in this respect so far. I would prefer to use tRPC only here. Any challenges in using that @Jonathan1599

Comment (Jonathan1599) [reply packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:17]: I'm not sure. I haven't used tRPC before. Is there an app integration that's currently using it? I could use it as a reference

Comment (hariombalhara) [reply packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:17]: Check this out https://github.com/calcom/cal.com/blob/1504715a0458f8d12e4f738ecb9eace9647287f3/packages/app-store/routing-forms/components/FormActions.tsx#L14

Comment (Jonathan1599) [reply packages/app-store/basecamp3/components/EventTypeAppCardInterface.tsx:17]: Thanks!

**Comment by leog** [comment]
> Yep, still working on it, thanks!

@Jonathan1599 Ok, gotcha. FYI the PR can become stale if there is no activity for a while.

**Comment by Jonathan1599** [comment]
Hey @leog could you take a look at this please? Or do I have to request a review from @hariombalhara instead? Not sure

**Review by leog — APPROVED** [review]
LGTM, great work! Thanks for contributing this!

**Pull request merged**
